src/core/stream/period/create_empty_adaptation_stream.ts

```diff
@@ -41,14 +41,18 @@ import { IStreamStatusEvent } from "../types";
 export default function createEmptyAdaptationStream(
   playbackObserver : IReadOnlyPlaybackObserver<{ position : number }>,
   wantedBufferAhead : IReadOnlySharedReference<number>,
+  maxVideoBufferSize : IReadOnlySharedReference<number>,
   bufferType : IBufferType,
   content : { period : Period }
 ) : Observable<IStreamStatusEvent> {
   const { period } = content;
   let hasFinishedLoading = false;
   const wantedBufferAhead$ = wantedBufferAhead.asObservable();
+  const maxVideoBufferSize$ = maxVideoBufferSize.asObservable();
   const observation$ = playbackObserver.observe(true);
-  return observableCombineLatest([observation$, wantedBufferAhead$]).pipe(
+  return observableCombineLatest([observation$,
+                                  wantedBufferAhead$,
+                                  maxVideoBufferSize$]).pipe(
```

Comment:
I think that there's no point relying on `maxVideoBufferSize` because it won't affect the inner Observable in any way.

The current file allows to create placeholder `AdaptationStream`s for type of buffer which are not actually loaded, such as the text buffer when there is no chosen subtitles or the video buffer when we are in audio-only.

Here they listen to `observation$` and `wantedBufferAhead$` because `AdaptationStream`s are supposed to send an event regularly (the "stream-status" event) and one of its property is about whether the end of that AdaptationStream's Period has been reached (the `hasFinishedLoading` property), so that we can create an `AdaptationStream` linked to the next Period for that type (this allows to pre-load future Period without conflicting with the loading logic of the current one).

The `hasFinishedLoading` property is here set to `true` as soon as the current position plus the wantedBufferAhead is reaching the end of that Period, as we're not loading any segment in that case.

But right now, I don't see any use for listening to the maxVideoBufferSize here